warehouse/query-core/src/main/java/datawave/query/iterator/logic/TermFrequencyExcerptIterator.java

```diff
@@ -276,66 +323,483 @@ public void next() throws IOException {
+        String phraseWithoutScoresExcerpt = generatePhrase(wordsAndScoresArr);
+        String oneBestExcerpt;
+
+        // if the regular excerpt is blank, we will return blank excerpts for the other ones as well.
+        if (phraseWithoutScoresExcerpt.isEmpty()) {
+            phraseWithScoresExcerpt = "";
+            oneBestExcerpt = "";
+        } else {
+            // if not scored, we won't output anything for this part
+            if (!usedScores && startOffset < endOffset) {
+                oneBestExcerpt = NOT_SCORED_MARKER;
+            } else {
+                // prepare all the WordsAndScores to output for the "one best" excerpt
+                for (WordsAndScores wordsAndScores : wordsAndScoresArr) {
+                    if (wordsAndScores == null) {
+                        continue;
+                    }
+                    wordsAndScores.setOneBestExcerpt(true);
+                }
+                // generate the "one best" excerpt
+                oneBestExcerpt = generatePhrase(wordsAndScoresArr);
+            }
+
+        }
+
+        // create a key that contains all of our excerpts to be read by the ExcerptTransform
+        tk = new Key(row, new Text(dtUid), new Text(
+                        fieldName + Constants.NULL + phraseWithScoresExcerpt + Constants.NULL + phraseWithoutScoresExcerpt + Constants.NULL + oneBestExcerpt),
+                        cv, ts);
         tv = new Value();
     }
 
     /**
-     * Generate a phrase from the given list of terms
+     * Generate a phrase from the given lists of WordsAndScores
      *
-     * @param terms
-     *            the terms to create a phrase from
+     * @param wordsAndScoresArr
+     *            the array of WordsAndScores that contain the terms to create a phrase from
      * @return the phrase
      */
-    protected String generatePhrase(List<String>[] terms) {
-        String[] largestTerms = new String[terms.length];
-        for (int i = 0; i < terms.length; i++) {
-            largestTerms[i] = getLongestTerm(terms[i]);
+    protected String generatePhrase(WordsAndScores[] wordsAndScoresArr) {
+        // put brackets around whole hit phrases instead of individual terms
+        checkForHitPhrase(wordsAndScoresArr);
+
+        // there are cases where we'll have no scores and we will want to choose a longer term for a particular position
+        // instead of the hit term we matched (e.g., when we're dealing with synonyms that are a fragment of a larger
+        // term.) This method will set an override for that position so that we'll choose the longer term instead of
+        // a shorter synonym, which is typically a substring of the longer term.
+        overrideOutputLongest(wordsAndScoresArr);
+
+        // create an array with the same length as the one we just passed in
+        String[] termsToOutput = new String[wordsAndScoresArr.length];
+        // TODO: what about both?
```

Comment:
"both" is the default so if a user does not specifically specify before/after, it will assume "both"